sympy/physics/continuum_mechanics/beam.py

```diff
@@ -209,6 +210,47 @@ def bc_deflection(self):
     def bc_deflection(self, d_bcs):
         self._boundary_conditions['deflection'] = d_bcs
 
+    def join(self, beam, via="fixed"):
+        """
+        This method joins two beams to make a new composite beam system.
+        Passed Beam class instance is attached to the right end of calling
+        object.
+
+        Examples
+        ========
+        There is a cantilever beam of length 4 meters. For first 2 meters
+        its moment of inertia is `1.5*I` and `I` for the other end.
+        A pointload of magnitude 4 N is applied from the top at its free end.
+
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import symbols
+        >>> E, I = symbols('E, I')
+        >>> R1, R2 = symbols('R1, R2')
+        >>> b1 = Beam(2, E, 1.5*I)
+        >>> b2 = Beam(2, E, I)
+        >>> b1.join(b2, "fixed")
```

Comment:
I'm not sure we should mutate b1. Would it be better to do:

```
b3 = b1.join(b2, 'fixed')
```

That way you can make use of b1 and b2 without worrying about them being mutated.